### PR TITLE
chore(docs): add custom signin page for Next.js client side

### PR DESCRIPTION
**Pull request by halvaradop**
## ☕️ Reasoning
This pull request adds a guide for creating a custom sign-in page for Next.js on the client side in the `guides/pages` section. The example includes:

- Code for handling credentials-based sign-in.
- Rendering of authentication providers dynamically.

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: docs/pages/guides/pages/signin.mdx [hunk 1]
```diff
@@ -192,6 +192,62 @@ export default async function SignInPage() {
 ```
 
 </Code.Next>
+<Code.NextClient>
+
+```tsx filename="app/signin/page.tsx"
+"use client"
+import { AuthError } from "next-auth"
+import { signIn } from "next-auth/react"
+import { providerMap } from "@/auth"
```
Comment (balazsorban44) [inline docs/pages/guides/pages/signin.mdx:201]: this is imported from a file with potentially server-side code, so most likely will break. <picture data-single-emoji=":sad-cowboy:" title=":sad-cowboy:">:sad-cowboy:</picture> I think to support this case, we would need to recommend a config split, which I don't think is worth it just to support client-side login pages, which don't really have much of a benefit anyway.

**Comment by balazsorban44** [comment]
Closing per https://github.com/nextauthjs/next-auth/pull/11793/files#r1760062143, but thank you anyway! :green_heart: lots of great contributions from you! :pray:

File: docs/pages/guides/pages/signin.mdx (hunk 1 above)
Comment (halvaradop) [reply docs/pages/guides/pages/signin.mdx:201]: That makes sense, it's similar to what was discussed in issue #11803. I might join the Discord and look into making some contributions as well.

**Comment by halvaradop** [comment]
Thank you for recognizing my work on this amazing project. I will continue working to improve my contributions! 👽

**Pull request closed without merging**